pkg/helm/controller/reconcile.go

```diff
@@ -240,7 +240,8 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 			r.EventRecorder.Eventf(o, "Warning", "OverrideValuesInUse",
 				"Chart value %q overridden to %q by operator's watches.yaml", k, v)
 		}
-		previousRelease, updatedRelease, err := manager.UpdateRelease(context.TODO())
+		force := o.GetAnnotations()["helm.sdk.operator-framework.io/upgrade-force"] == "true"
+		previousRelease, updatedRelease, err := manager.UpdateRelease(context.TODO(), release.ForceUpdate(force))
```

Comment:
We should convert the string to bool value and then check it instead of comparing a string. Otherwise, it will not work if the user passes True or true for example. Following the  suggestion:

```suggestion
		previousRelease, updatedRelease, err := manager.UpdateRelease(context.TODO(), release.ForceUpdate(hasHelmUpgradeForceAnnotation()))
```
```go
// hasHelmUpgradeForceAnnotation will return true when the annotation to allow force resources replacement 
// (`helm upgrade --force`) will be informed
func hasHelmUpgradeForceAnnotation() bool {
         const helmUpgradeForceAnnotation = "helm.operator-sdk/upgrade-force"
         force := o.GetAnnotations()[helmUpgradeForceAnnotation]
         value := false   
	if i, err := strconv.ParseBool(HelmUpgradeForceAnnotation); err != nil {
		log.Info("Could not parse annotation as an boolean",
			"annotation", helmUpgradeForceAnnotation, "value informed", force)
	} else {
		value = i
	}
	return value
}
```

Reply:
changed: i'm not printing the log info if the annotation is empty/not set